File: backend/app/schemas/config_sections/chapter_reward_config.py
from pydantic import BaseModel, Field


class ChapterRewardConfig(BaseModel):
    """Chapter completion reward configuration"""
    Undo: int = Field(ge=0, description="Number of Undo boosters rewarded")
    Hint: int = Field(ge=0, description="Number of Hint boosters rewarded")
    Shuffle: int = Field(ge=0, description="Number of Shuffle boosters rewarded")
    

File: frontend/src/lib/validations/chapterRewardConfig.ts
import { z } from 'zod';

export const chapterRewardConfigSchema = z.object({
  undo: z.number().int().min(0, 'Undo reward must be 0 or greater'),
  hint: z.number().int().min(0, 'Hint reward must be 0 or greater'),
  shuffle: z.number().int().min(0, 'Shuffle reward must be 0 or greater'),
});

export type ChapterRewardConfig = z.infer<typeof chapterRewardConfigSchema>;

export const defaultChapterRewardConfig: ChapterRewardConfig = {
  undo: 0,
  hint: 0,
  shuffle: 0,
};

